lib/models/Thread.js

```diff
@@ -47,6 +47,88 @@ export default class Thread extends Model {
   }
 
   /**
+   * Updates the isFlagged field of all {@link Message}s of this {@link Thread}.
+   *
+   * @param isFlagged {Boolean} The isFlagged field of the thread.
+   *
+   * @return {Promise} A {@link Promise}, as per {@link Client.setMessages}.
+   *
+   * @see Client#setMessages
+   */
+  setIsFlagged(isFlagged) {
+    return this._flagAllMessages('isFlagged', isFlagged);
+  }
+
+  /**
+   * Updates the isUnread field of this {@link Message}.
+   *
+   * @param isUnread {Boolean} The isUnread field of the thread.
+   *
+   * @return {Promise} A {@link Promise}, as per {@link Client.setMessages}.
+   *
+   * @see Client#setMessages
+   */
+  setIsUnread(isUnread) {
+    return this._flagAllMessages('isUnread', isUnread);
+  }
+
+  /**
+   * Moves this {@link Thread} to a different set of mailboxes.
+   *
+   * @param mailboxIds {String[]} The identifiers of the target mailboxes for the thread.
+   *
+   * @return {Promise} A {@link Promise}, as per {@link Client.setMessages}.
+   *
+   * @see Client#setMessages
+   */
+  move(mailboxIds) {
+    return this._updateAllMessages({ mailboxIds });
```

Comment:
So I have an idea:
We have said that Thread is a virtual concept while the real one is the messages. So I propose to add updateAllMessages and flagAllMessages in the Message model and we can leverage them from here. Cause it is not easy for a normal user to understand why do we have such operationAllMessages in the Thread model, WDYT?